src/components/settings/SettingsLMSTab/tests/CanvasConfig.test.jsx

```diff
@@ -148,24 +184,120 @@ describe('<CanvasConfig />', () => {
+        target: { value: 'test2' },
+      });
+      fireEvent.change(screen.getByLabelText('Canvas Account Number'), {
+        target: { value: '3' },
+      });
+      fireEvent.change(screen.getByLabelText('Display Name'), {
+        target: { value: 'displayName' },
+      });
+      fireEvent.change(screen.getByLabelText('Canvas Base URL'), {
+        target: { value: 'https://www.test4.com' },
+      });
+    });
+    expect(screen.getByText('Authorize')).not.toBeDisabled();
+    await act(async () => {
+      fireEvent.click(screen.getByText('Authorize'));
+      fireEvent.click(screen.getByText('Cancel'));
+    });
+    await act(async () => {
+      fireEvent.click(screen.getByText('Save'));
     });
-    fireEvent.click(screen.getByText('Cancel'));
-    fireEvent.click(screen.getByText('Save'));
     const expectedConfig = {
       active: false,
       display_name: 'displayName',
       enterprise_customer: enterpriseId,
+      canvas_account_id: '3',
+      canvas_base_url: 'https://www.test4.com',
+      client_id: 'test1',
+      client_secret: 'test2',
     };
     expect(LmsApiService.postNewCanvasConfig).toHaveBeenCalledWith(expectedConfig);
   });
+  test('Authorizing a config will initial backend polling', async () => {
+    render(
+      <CanvasConfig
+        enterpriseCustomerUuid={enterpriseId}
+        onClick={mockOnClick}
+        existingData={noExistingData}
+      />,
+    );
+    await act(() => {
+      fireEvent.change(screen.getByLabelText('Display Name'), {
+        target: { value: 'displayName' },
+      });
+      fireEvent.change(screen.getByLabelText('Canvas Base URL'), {
+        target: { value: 'https://www.test.com' },
+      });
+      fireEvent.change(screen.getByLabelText('API Client ID'), {
+        target: { value: 'test1' },
+      });
+      fireEvent.change(screen.getByLabelText('API Client Secret'), {
+        target: { value: 'test2' },
+      });
+      fireEvent.change(screen.getByLabelText('Canvas Account Number'), {
+        target: { value: '3' },
+      });
+    });
+    expect(screen.getByText('Authorize')).not.toBeDisabled();
+    await act(async () => {
+      fireEvent.click(screen.getByText('Authorize'));
+    });
+    await screen.findByText('Display Name');
+    expect(mockFetchSingleConfig).toHaveBeenCalledWith(1);
+  });
+  test('Authorizing an existing, edited config will call update config endpoint', async () => {
+    render(
+      <CanvasConfig
+        enterpriseCustomerUuid={enterpriseId}
+        onClick={mockOnClick}
+        existingData={existingConfigDataNoAuth}
+      />,
+    );
+    act(() => {
+      fireEvent.change(screen.getByLabelText('Display Name'), {
+        target: { value: 'displayName' },
+      });
+      fireEvent.change(screen.getByLabelText('Canvas Base URL'), {
+        target: { value: 'https://www.test.com' },
+      });
+    });
+    expect(screen.getByText('Authorize')).not.toBeDisabled();
+    await act(async () => {
```

Comment:
yeah I am certain in genearl we don't need act around the event actions, esp if you use user-events library

you want to await act() on whatever needs to be awaited for _after_ you click that always works

Reply:
because of the hooks we actually need the acts

Reply:
to be more descriptive, I was getting act errors when not wrapping these, I believe because we have hooks running in the background

Reply:
hmm ok we can leave these be for now then, maybe I missed something. Generally the event triggering functions if you use the user-events library are already wrapped in act. I am guessing you need to do this because you are using the built in fireEvent instead of user-events?